src/main/java/com/devcycle/sdk/server/local/api/DVCLocalClient.java

```diff
@@ -0,0 +1,167 @@
+package com.devcycle.sdk.server.local.api;
+
+import java.util.Collections;
+import java.util.Map;
+
+import com.devcycle.sdk.server.common.model.*;
+import com.devcycle.sdk.server.local.bucketing.LocalBucketing;
+import com.devcycle.sdk.server.local.managers.EnvironmentConfigManager;
+import com.devcycle.sdk.server.local.managers.EventQueueManager;
+import com.devcycle.sdk.server.local.model.BucketedUserConfig;
+import com.devcycle.sdk.server.local.model.DVCLocalOptions;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public final class DVCLocalClient {
+
+  private static LocalBucketing localBucketing = new LocalBucketing();
+
+  private EnvironmentConfigManager configManager;
+
+  private final String serverKey;
+
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+  private EventQueueManager eventQueueManager;
+
+  public DVCLocalClient(String serverKey) {
+    this(serverKey, DVCLocalOptions.builder().build());
+  }
+
+  public DVCLocalClient(String serverKey, DVCLocalOptions dvcOptions) {
+    configManager = new EnvironmentConfigManager(serverKey, localBucketing, dvcOptions);
+    this.serverKey = serverKey;
+    OBJECT_MAPPER.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+    try {
+      eventQueueManager = new EventQueueManager(serverKey, localBucketing, dvcOptions);
+    } catch (Exception e) {
+      System.out.printf("Error creating event queue due to error: %s%n", e.getMessage());
+    }
+  }
+
+  /**
+   * Get all features for user data
+   * 
+   * @param user (required)
+   */
+  public Map<String, Feature> allFeatures(User user) {
+    validateUser(user);
+    localBucketing.setPlatformData(user.getPlatformData().toString());
+    String userString = null;
+    BucketedUserConfig bucketedUserConfig = null;
+
+    try {
+      userString = OBJECT_MAPPER.writeValueAsString(user);
+
+      bucketedUserConfig = localBucketing.generateBucketedConfig(serverKey, userString);
```

Comment:
if this is a common action, might make sense to move this logic to the `generateBucketedConfig` method.